zp-relayer/workers/sentTxWorker.ts

```diff
@@ -85,15 +49,7 @@ export async function createSentTxWorker<T extends EstimationType>(gasPrice: Gas
   const sentTxWorkerProcessor = async (job: Job<SentTxPayload>) => {
     const logPrefix = `SENT WORKER: Job ${job.id}:`
     logger.info('%s processing...', logPrefix)
-    const { txType, txHash, prefixedMemo, commitIndex, outCommit, nullifier, root, txData } = job.data
-
-    // TODO: it is possible that a tx marked as failed could be stuck
-    // in the mempool. Worker should either assure that it is mined
-    // or try to substitute such transaction with another one
-    if (await checkMarked(redis, job.id as string)) {
-      logger.info('%s marked as failed, skipping', logPrefix)
-      return [SentTxState.REVERT, txHash] as SentTxResult
-    }
+    const { txHash, prefixedMemo, commitIndex, outCommit, nullifier, root } = job.data
 
     const tx = await web3.eth.getTransactionReceipt(txHash)
```

Comment:
I think this line should be a little bit more complex than that...

For each job X, we might have multiple transactions with same nonce, but different gas prices. We can't rely on and check only the last sent one. We should check all their statuses, in the gas price descending order, until the first one mined is found.

To reduce number of queries on each iteration we could start by querying the `eth_getTransactionCount` first. If actual nonce is less than or equal to the transaction nonce, then none of them were mined for sure.

If the actual nonce is greater than the transaction nonce, but none of the searched transaction hashes lead to a valid result, then we should move on to the next job, leaving the current one without attached tx hash and re-sending it back into the poolTxWorker for re-validation (this weird situation should only happen if someone will manually send a transaction on behalf of the relayer).

Reply:
Isn’t it guaranteed that if `eth_sendRawTransaction` call was successful, then submitted tx will replace previous one (with the same nonce) in the mempool?

Reply:
The mempool is distributed among validator nodes, there always will be a race condition between its update and actual block proposals. So there might be some rare cases when older transaction will still be included, and new one will be rejected.

Afaik, in the current version it will lead to a dead loop of checking transaction status, thus we should make some workaround.

Reply:
Agree, will add an update